### PR TITLE
mc6809: update to match new RasmOp API

### LINKED ISSUE #167: compilation of the mc6809 module fails
from a fresh radare2 install in a docker (via a docker pull):
```
r2@radare:~$ r2pm -i mc6809
Already up-to-date.
Install Done For mc6809
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu
checking for working directories... current
using prefix '/home/r2/.local/share/radare2/prefix'
checking for c compiler... gcc
Using PKGCONFIG: pkg-config
checking pkg-config flags for r_core... yes
checking for libewf ... no
checking pkg-config flags for libewf... no
checking pkg-config flags for unicorn... no
checking for libyara ... no
checking pkg-config flags for yara... no
creating ./Makefile
creating ./options.mk
cleaning temporally files... done

Final report:
 - PREFIX = /home/r2/.local/share/radare2/prefix
rm -f *.so *.o asm_x86_udis.o asm_x86_olly.o ../arch/x86/ollyasm/disasm.o ../arch/x86/ollyasm/asmserv.o ../arch/x86/ollyasm/assembl.o asm_x86_tab.o asm_armthumb.o ../arch/arm/armthumb.o ../arch/ppc/ppc_disasm/ppc_disasm.o asm_ppc_disasm.o asm_psosvm.o ../arch/psosvm/vmas/vmas.o asm_m68k_net.o ../arch/m68k/m68k_disasm/m68k_disasm.o asm_blackfin.o asm_x86_bea.o ../arch/x86/bea/BeaEngine.o asm_msil.o asm_nios.o ../arch/nios/gnu/safe-ctype.o ../arch/nios/gnu/cgen-bitset.o ../arch/nios/gnu/cgen-dis.o ../arch/nios/gnu/cgen-asm.o ../arch/nios/gnu/cgen-opc.o ../arch/nios/gnu/cpu-nios.o ../arch/nios/gnu/nios-dis.o ../arch/nios/gnu/nios-asm.o ../arch/nios/gnu/nios-desc.o ../arch/nios/gnu/nios-ibld.o ../arch/nios/gnu/nios-opc.o asm_swf.o ../arch/swf/swfdis.o asm_vc4.o ../arch/vc4/vc4-dis.o ../arch/vc4/vc4-asm.o ../arch/vc4/vc4-desc.o ../arch/vc4/vc4-ibld.o ../arch/vc4/vc4-opc.o ../arch/vc4/cgen-bitset.o ../arch/vc4/cgen-dis.o ../arch/vc4/cgen-asm.o ../arch/vc4/cgen-opc.o ../arch/vc4/libvc4/eval.o ../arch/vc4/libvc4/decode.o ../arch/vc4/libvc4/vc4_decode.o ../arch/vc4/libvc4/vc4_arch.o ../arch/vc4/libvc4/vc4_util.o asm_mc6809.o ../arch/z80_nc/z80.o ../arch/z80_nc/disasm.o ../arch/z80_nc/test.o asm_z80_nc.o asm_pyc.o ../arch/pyc/pyc_disasm.o asm_x86_zyan.o ../arch/x86/zyan/src/Decoder.o ../arch/x86/zyan/src/Formatter.o ../arch/x86/zyan/src/InstructionTable.o ../arch/x86/zyan/src/Mnemonic.o ../arch/x86/zyan/src/Register.o ../arch/x86/zyan/src/Utils.o ../arch/x86/zyan/src/Zydis.o asm_evm.o ../arch/evm/evm.o asm_atombios.o ../arch/atombios/atombios.o
Makefile	asm_mc6809.c	  asm_vc4.c	  asm_z80_nc.c	msil.mk        x86_bea.mk
armthumb.mk	asm_msil.c	  asm_x86_bea.c   atombios.mk	nios.mk        x86_olly.mk
asm_armthumb.c	asm_nios.c	  asm_x86_olly.c  blackfin.mk	ppc_disasm.mk  x86_tab.mk
asm_atombios.c	asm_ppc_disasm.c  asm_x86_tab.c   configure	psosvm.mk      x86_udis.mk
asm_blackfin.c	asm_psosvm.c	  asm_x86_udis.c  evm.mk	pyc.mk	       x86_zyan.mk
asm_evm.c	asm_pyc.c	  asm_x86_zyan.c  m68k_net.mk	swf.mk	       z80.mk
asm_m68k_net.c	asm_swf.c	  asm_z80.c	  mc6809.mk	vc4.mk	       z80_nc.mk
gcc -I/home/r2/.local/share/radare2/prefix/include -I/usr/include/libr -g -I/shlr -I../../include -I../arch/ -I../arch/include -Wall -shared -fPIC  -DLIL_ENDIAN=1 -I../arch/x86 -I../arch/x86/bea/ -I../arch/x86/bea/src -I../arch/x86/bea/include -DUint64=uint64_t -I../arch/swf -Iarch/vc4/libvc4/ -Iarch/include/ -I../arch/pyc -I../arch/x86/zyan/include -I../arch/evm   -c -o asm_mc6809.o asm_mc6809.c
asm_mc6809.c: In function 'mc6809_disassemble':
asm_mc6809.c:1069:11: error: incompatible type for argument 1 of 'strcpy'
   strcpy (op->buf_asm, "INVALID");
           ^~
In file included from /usr/include/libr/r_types.h:168:0,
                 from asm_mc6809.c:4:
/usr/include/string.h:125:14: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern char *strcpy (char *__restrict __dest, const char *__restrict __src)
              ^~~~~~
asm_mc6809.c:1073:11: error: incompatible type for argument 1 of 'strcpy'
   strcpy (op->buf_asm, mc6809_opcode->name);
           ^~
In file included from /usr/include/libr/r_types.h:168:0,
                 from asm_mc6809.c:4:
/usr/include/string.h:125:14: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern char *strcpy (char *__restrict __dest, const char *__restrict __src)
              ^~~~~~
asm_mc6809.c:1077:12: error: incompatible type for argument 1 of 'sprintf'
   sprintf (op->buf_asm,
            ^~
In file included from asm_mc6809.c:3:0:
/usr/include/stdio.h:366:12: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern int sprintf (char *__restrict __s,
            ^~~~~~~
asm_mc6809.c:1082:12: error: incompatible type for argument 1 of 'sprintf'
   sprintf (op->buf_asm,
            ^~
In file included from asm_mc6809.c:3:0:
/usr/include/stdio.h:366:12: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern int sprintf (char *__restrict __s,
            ^~~~~~~
asm_mc6809.c:1088:12: error: incompatible type for argument 1 of 'sprintf'
   sprintf (op->buf_asm,
            ^~
In file included from asm_mc6809.c:3:0:
/usr/include/stdio.h:366:12: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern int sprintf (char *__restrict __s,
            ^~~~~~~
asm_mc6809.c:1093:12: error: incompatible type for argument 1 of 'sprintf'
   sprintf (op->buf_asm, "%s $%04x",
            ^~
In file included from asm_mc6809.c:3:0:
/usr/include/stdio.h:366:12: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern int sprintf (char *__restrict __s,
            ^~~~~~~
asm_mc6809.c:1099:12: error: incompatible type for argument 1 of 'sprintf'
   sprintf (op->buf_asm, "%s $%04x", mc6809_opcode->name,
            ^~
In file included from asm_mc6809.c:3:0:
/usr/include/stdio.h:366:12: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern int sprintf (char *__restrict __s,
            ^~~~~~~
asm_mc6809.c:1108:12: error: incompatible type for argument 1 of 'strcpy'
    strcpy (op->buf_asm, "INVALID");
            ^~
In file included from /usr/include/libr/r_types.h:168:0,
                 from asm_mc6809.c:4:
/usr/include/string.h:125:14: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern char *strcpy (char *__restrict __dest, const char *__restrict __src)
              ^~~~~~
asm_mc6809.c:1116:13: error: incompatible type for argument 1 of 'strcpy'
     strcpy (op->buf_asm, "INVALID");
             ^~
In file included from /usr/include/libr/r_types.h:168:0,
                 from asm_mc6809.c:4:
/usr/include/string.h:125:14: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern char *strcpy (char *__restrict __dest, const char *__restrict __src)
              ^~~~~~
asm_mc6809.c:1119:14: error: incompatible type for argument 1 of 'sprintf'
     sprintf (op->buf_asm,
              ^~
In file included from asm_mc6809.c:3:0:
/usr/include/stdio.h:366:12: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern int sprintf (char *__restrict __s,
            ^~~~~~~
asm_mc6809.c:1130:11: error: incompatible type for argument 1 of 'strcpy'
   strcpy (op->buf_asm, mc6809_opcode->name);
           ^~
In file included from /usr/include/libr/r_types.h:168:0,
                 from asm_mc6809.c:4:
/usr/include/string.h:125:14: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern char *strcpy (char *__restrict __dest, const char *__restrict __src)
              ^~~~~~
asm_mc6809.c:1131:43: error: incompatible type for argument 1 of 'mc6809_append_indexed_args'
   op->size += mc6809_append_indexed_args (op->buf_asm,
                                           ^~
asm_mc6809.c:830:12: note: expected 'char *' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 static int mc6809_append_indexed_args (char *buf_asm, const ut8 *buf)
            ^~~~~~~~~~~~~~~~~~~~~~~~~~
asm_mc6809.c:1136:11: error: incompatible type for argument 1 of 'strcpy'
   strcpy (op->buf_asm, mc6809_opcode->name);
           ^~
In file included from /usr/include/libr/r_types.h:168:0,
                 from asm_mc6809.c:4:
/usr/include/string.h:125:14: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern char *strcpy (char *__restrict __dest, const char *__restrict __src)
              ^~~~~~
asm_mc6809.c:1138:8: error: incompatible type for argument 2 of 'mc6809_append_pushpull_args'
        op->buf_asm,
        ^~
asm_mc6809.c:999:12: note: expected 'char *' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 static int mc6809_append_pushpull_args(enum instruction_mode mode,
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~
asm_mc6809.c:1142:12: error: incompatible type for argument 1 of 'sprintf'
   sprintf (op->buf_asm,
            ^~
In file included from asm_mc6809.c:3:0:
/usr/include/stdio.h:366:12: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern int sprintf (char *__restrict __s,
            ^~~~~~~
<builtin>: recipe for target 'asm_mc6809.o' failed
make: *** [asm_mc6809.o] Error 1

```

### PR DESCRIPTION
**Pull request by douardda**
A first attempt to fix #167

### CODE REVIEW

File: libr/asm/p/asm_mc6809.c [hunk 1]
```diff
@@ -992,50 +992,51 @@ static int mc6809_append_indexed_args (char *buf_asm, const ut8 *buf)
 			}
 		}
 	}
-	strcat (buf_asm, postop_buffer);
+	r_strbuf_append(buf_asm, postop_buffer);
```
Comment (radare) [inline libr/asm/p/asm_mc6809.c]: space before(

File: libr/asm/p/asm_mc6809.c [hunk 2]
```diff
@@ -827,7 +827,7 @@ static const char mc6809_index_registers[] = {
 	/* 0b11 */ 's',
 };
 
-static int mc6809_append_indexed_args (char *buf_asm, const ut8 *buf)
+static int mc6809_append_indexed_args (RStrBuf *buf_asm, const ut8 *buf)
```
Comment (radare) [inline libr/asm/p/asm_mc6809.c]: remove space before ( and open brace { in this line

File: libr/asm/p/asm_mc6809.c [hunk 3]
```diff
@@ -992,50 +992,51 @@ static int mc6809_append_indexed_args (char *buf_asm, const ut8 *buf)
...
 	return postop_bytes;
 }
 
 static int mc6809_append_pushpull_args(enum instruction_mode mode,
-				       char *buf_asm,
+				       RStrBuf *buf_asm,
 				       const ut8 *opcode_args)
 {
-	strcat (buf_asm, " ");
+	r_strbuf_append(buf_asm, " ");
```
Comment (radare) [inline libr/asm/p/asm_mc6809.c]: missing space before (

File: libr/asm/p/asm_mc6809.c [hunk 4]
```diff
@@ -992,50 +992,51 @@ static int mc6809_append_indexed_args (char *buf_asm, const ut8 *buf)
...
+	r_strbuf_get(buf_asm)[r_strbuf_length(buf_asm)-1] = '\0';
 	return 2;
 }
 
 static int mc6809_disassemble(RAsm *a, RAsmOp *op, const ut8 *buf, int len) {
 	ut8 tfrexg_regmasked;
 	const char *tfrexg_source_reg;
 	const char *tfrexg_dest_reg;
-
+        const char *buf_asm = "invalid";
```
Comment (radare) [inline libr/asm/p/asm_mc6809.c]: use tabs

File: libr/asm/p/asm_mc6809.c [hunk 5]
```diff
@@ -1105,18 +1102,16 @@ static int mc6809_disassemble(RAsm *a, RAsmOp *op, const ut8 *buf, int len) {
...
 				mc6809_register_field[(*opcode_args >> 4) & 0x0f];
 			tfrexg_dest_reg = \
 				mc6809_register_field[*opcode_args & 0x0f];
 			if (!tfrexg_source_reg || !tfrexg_dest_reg) {
 				op->size += 1;
-				strcpy (op->buf_asm, "INVALID");
 			} else {
 				op->size += 2;
-				sprintf (op->buf_asm,
+				buf_asm = sdb_fmt(
```
Comment (radare) [inline libr/asm/p/asm_mc6809.c]: missing space

**Comment by douardda** [comment]
I've fixed (hopefully) all the spacing/indentation problems.

**Pull request merged**